common/test/acceptance/tests/lms/test_lms_edxnotes.py

```diff
@@ -536,6 +535,57 @@ def test_no_content(self):
             "You have not made any notes in this course yet. Other students in this course are using notes to:",
             notes_page_empty.no_content_text)
 
+    def test_notes_works_correctly_with_xss(self):
+        """
+        Scenario: Note text & tags should be HTML and JS escaped
+        Given I am enrolled in a course with notes enabled
+        When I visit the Notes page, with a Notes text & tag containing JS code
+        Then I should see them escaped.
```

Comment:
Thanks @muzaffaryousaf.  I mentioned I was in a rush. :)  I'm not sure why I thought this was a Python test, but either way, I'd still like to understand how this works and if it works.

The first tag would ultimately be HTML-escaped as follows:
```
&lt;script&gt;alert(&#34;XSS&#34;)&lt;/script&gt;
```
The assertion does not compare against this, but against the original text.  So, one of the following is happening:
1. This isn't a proper test and the escaping isn't happening. (Note, I do not think this is the case.) Or,
2. Something in the way the page object works is turning the results of `self.notes_page.notes` back into text without the escaping, or
3. Something else I'm missing.

Would you be able to clarify the following:
1. If you remove the final HTML-escaping from the actual code, does this test fail?  In other words, does it catch what it says it will catch?
2. Can you help me understand how it works?  I simply don't get how this works if it works.

Thank you!